File: frontend/src/api/admin/knowledgeBase.ts
import apiClient from '../client'

export interface KnowledgeBase {
  id: string
  name: string
  description: string
  show_in_lab: boolean
  ai_searchable: boolean
  is_enabled: boolean
  doc_count: number
  updated_at: string | null
}

export interface KnowledgeBaseDoc {
  doc_id: string
  title: string
  category: string
  status: number
  updated_at: string | null
}

export const knowledgeBaseApi = {
  async list(): Promise<KnowledgeBase[]> {
    const { data } = await apiClient.get('/admin/knowledge-bases')
    return data || []
  },
  async create(payload: Partial<KnowledgeBase>): Promise<KnowledgeBase> {
    const { data } = await apiClient.post('/admin/knowledge-bases', payload)
    return data
  },
  async update(id: string, payload: Partial<KnowledgeBase>): Promise<KnowledgeBase> {
    const { data } = await apiClient.put(`/admin/knowledge-bases/${id}`, payload)
    return data
  },
  async remove(id: string): Promise<void> {
    await apiClient.delete(`/admin/knowledge-bases/${id}`)
  },
  async docs(id: string): Promise<KnowledgeBaseDoc[]> {
    const { data } = await apiClient.get(`/admin/knowledge-bases/${id}/docs`)
    return data || []
  },
}

File: src/cygnusx/application/schemas/knowledge_base.py
"""知识库资源管理 DTO。"""

from datetime import datetime

from pydantic import BaseModel, Field


class KnowledgeBaseDTO(BaseModel):
    id: str
    project_id: str | None = None
    name: str
    description: str = ""
    show_in_lab: bool = True
    ai_searchable: bool = True
    is_enabled: bool = True
    doc_count: int = 0
    updated_at: datetime | None = None


class KnowledgeBaseCreateDTO(BaseModel):
    id: str = Field(min_length=1, max_length=64, pattern=r"^[a-z0-9][a-z0-9-]*$")
    project_id: str | None = Field(None, max_length=64)
    name: str = Field(min_length=1, max_length=128)
    description: str = ""
    show_in_lab: bool = True
    ai_searchable: bool = True


class KnowledgeBaseUpdateDTO(BaseModel):
    project_id: str | None = Field(None, max_length=64)
    name: str = Field(min_length=1, max_length=128)
    description: str = ""
    show_in_lab: bool = True
    ai_searchable: bool = True
    is_enabled: bool = True


class KnowledgeBaseDocDTO(BaseModel):
    doc_id: str
    title: str
    category: str
    status: int
    updated_at: datetime | None = None
